inc/generation.class.php

```diff
@@ -95,8 +95,8 @@ static function autoName($config, CommonDBTM $item) {
             date('Y'),
             date('m'),
             date('d'),
-            $item->fields['serial'] ?? '',
-            $item->fields['name'] ?? '',
+            $item->input['serial'] ?? '',
+            $item->input['name'] ?? '',
```

Comment:
Generation could also be done on existing objects, using massive actions.

```suggestion
            $item->input['serial'] ?? $item->fields['serial'] ?? '',
            $item->input['name'] ?? $item->fields['name'] ?? '',
```